src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -15,410 +15,845 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/* ScriptData
-SDName: Argent Challenge Encounter.
-SD%Complete: 50 %
-SDComment: AI for Argent Soldiers are not implemented. AI from bosses need more improvements.
-SDCategory: Trial of the Champion
-EndScriptData */
-
+#include "Player.h"
 #include "ScriptMgr.h"
-#include "InstanceScript.h"
-#include "MotionMaster.h"
-#include "ObjectAccessor.h"
-#include "ScriptedEscortAI.h"
+#include "PassiveAI.h"
+#include "ScriptedCreature.h"
 #include "SpellScript.h"
-#include "TemporarySummon.h"
 #include "trial_of_the_champion.h"
-/*
+
 enum Yells
 {
     // Eadric the Pure
-    SAY_INTRO                   = 0,
-    SAY_AGGRO                   = 1,
-    EMOTE_RADIANCE              = 2,
-    EMOTE_HAMMER_RIGHTEOUS      = 3,
-    SAY_HAMMER_RIGHTEOUS        = 4,
-    SAY_KILL_PLAYER             = 5,
-    SAY_DEFEATED                = 6,
+    SAY_AGGRO_E = 1,
+    EMOTE_RADIANCE = 2,
+    EMOTE_HAMMER_RIGHTEOUS = 3,
+    SAY_HAMMER_RIGHTEOUS = 4,
+    SAY_KILL_UNIT_E = 5,
+    SAY_DEFEATED_E = 6,
 
     // Argent Confessor Paletress
-    SAY_INTRO_1                 = 0,
-    SAY_INTRO_2                 = 1,
-    SAY_AGGRO                   = 2,
-    SAY_MEMORY_SUMMON           = 3,
-    SAY_MEMORY_DEATH            = 4,
-    SAY_KILL_PLAYER             = 5,
-    SAY_DEFEATED                = 6,
+    SAY_AGGRO_P = 2,
+    SAY_MEMORY_SUMMON = 3,
+    SAY_MEMORY_DEATH = 4,
+    SAY_KILL_UNIT_P = 5,
+    SAY_DEFEATED_P = 6,
 
     // Memory of X
-    EMOTE_WAKING_NIGHTMARE      = 0
+    EMOTE_WAKING_NIGHTMARE = 0
 };
-*/
+
+enum Events
+{
+    // Argent Lightwielder
+    EVENT_BLAZING_LIGHT = 1,
+    EVENT_CLEAVE,
+    EVENT_UNBALANCING_STRIKE,
+
+    // Argent Monk
+    EVENT_FLURRY_OF_BLOWS,
+    EVENT_PUMMEL,
+
+    // Argent Priestess
+    EVENT_HOLY_SMITE,
+    EVENT_SHADOW_WORD_PAIN,
+    EVENT_FOUNTAIN,
+    EVENT_MIND_CONTROL,
+
+    // Argent Confessor Paletress
+    EVENT_HOLY_FIRE,
+    EVENT_HOLY_SMITE_E,
+    EVENT_RENEW,
+    EVENT_SUMMON_MEMORY,
+    EVENT_MEMORY_AGGRESSIVE,
+
+    // Memory
+    EVENT_OLD_WOUNDS,
+    EVENT_SHADOWS_PAST,
+    EVENT_WAKING_NIGHTMARE,
+
+    // Eadric the Pure
+    EVENT_RADIANCE,
+    EVENT_HAMMER_OF_JUSTICE,
+
+    // Shared
+    EVENT_CHALLENGE_DONE
+};
+
 enum Spells
 {
+    // Argent Lightwielder
+    SPELL_BLAZING_LIGHT = 67247,
+    SPELL_CLEAVE = 15284,
+    SPELL_UNBALANCING_STRIKE = 67237,
```

Comment:
Cosmetic/codestyle: align these values (like they used to be in the old code) when your PR gets close to ready for merging (also further down in this file and the other files).